[WWST-2446, WWST-2447] New DTH for Qubino Flush 1 Relay,  Qubino Flush 1D Relay

MGoralczykS opened the pull request:
I used child DTH's from @BJanuszS PR.

This Qubino supports Qubino Temperature Sensor ZMNHEA1 (as a child device) - temperature sensor only works with some Qubino devices. 

@greens @dkirker @tpmanley @KKlimczukS @PKacprowiczS @ZWozniakS @MWierzbinskaS @BJanuszS 
Could you review this?

MGoralczykS committed:
New DTH for Qubino Flush 1D Relay

MGoralczykS committed:
Delete unncessesary minHubCoreVersion from definition

greens commented:
I think this will be on hold until we decide the correct way to proceed with https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/30326 . I don't think the model being used is the correct way forward.

MGoralczykS committed:
Remove few settings

MGoralczykS commented:
@greens I removed some settings. I discussed them with @KKlimczukS 
Also, there is one setting that needs to be discussed. 
I mean setting #100 Enable / Disable Endpoint I2 or select Notification Type and Event 
(https://products.z-wavealliance.org/products/1336). Should I add this? This setting occurs in other relays too.
What do you think about this?

greens commented:
We're not going to be using parameter 100.

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,378 @@
+		)
+
+		switch(it.type) {
+			case "boolRange":
```
this device has no "boolRange" parameter and a lot of the boilerplate is dedicated to it.

Pare this list down to the actual types included in this PR and also inline most of this code

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,378 @@
+	updateDataValue("MSR", msr)
+	updateDataValue("manufacturer", cmd.manufacturerName)
+	createEvent([descriptionText: "$device.displayName MSR: $msr", isStateChange: false])
+}
```
this code is obsolete

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,378 @@
+}
+
+def installed() {
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```
should not be offlinePingable

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,378 @@
+	commands << zwave.basicV1.basicGet().format()
+	if (getDataValue("MSR") == null) {
+		commands << zwave.manufacturerSpecificV1.manufacturerSpecificGet().format()
+	}
```
remove

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,378 @@
+
+private notNullCheck(value) {
+	return value != null
+}
```
should not be a separate function

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,378 @@
+	def commands = []
+	parameterMap.each {
+		if (state.currentPreferencesState."$it.key".status == "syncPending") {
+            commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: getCommandValue(it), parameterNumber: it.parameterNumber, size: it.size))
```
nit: spacing

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,378 @@
+		parameterNumber: 63, size: 1, defaultValue: 0,
+		optionInactive: 0, inactiveDescription: "When system is turned off the output is 0V (NC).",
+		optionActive: 1, activeDescription: "When system is turned off the output is 230V or 24V (NO).",
+		description: "Support output Switch selection Set value means the type of the device that is connected to the output. The device type can be normally open (NO) or normally close (NC).  "
```
"Set the type of device connected to the output"

MGoralczykS committed:
Refactoring, added temperature event handling

MGoralczykS commented:
@greens I changed those things that you mentioned and I added events which are handling temperature.

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,402 @@
+
+def configure() {
+	def commands = []
+	commands << zwave.manufacturerSpecificV2.manufacturerSpecificGet().format()
```
remove this get

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,402 @@
+	commands << zwave.associationV1.associationSet(groupingIdentifier:3, nodeId:[zwaveHubNodeId]).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:4, nodeId:[zwaveHubNodeId]).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:5, nodeId:[zwaveHubNodeId]).format()
+	commands << zwave.associationV1.associationSet(groupingIdentifier:6, nodeId:[zwaveHubNodeId]).format()
```
can you comment what each association group is responsible for?

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,402 @@
+		def childDthNamespace 	= "qubino"
+
+		createChildDevice(childDthNamespace, childDthName, childDni, childComponentLabel)
+		childDevice = childDevices.find { it.deviceNetworkId == childDni }
```
you return the child device from `createChildDevice` you don't need to search for it

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,402 @@
+	delayBetween(commands)
+}
+
+def createChildDevice(childDthNamespace, childDthName, childDni, childComponentLabel) {
```
you only ever call this one time with one set of arguments, you should probably just hard-code them inside the function

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,402 @@
+	log.debug "on"
+	delayBetween([
+			zwave.basicV1.basicSet(value: 0xFF).format(),
+			zwave.basicV1.basicGet().format()
```
(here and several other places) Why are you only occasionally using `encap`? You _should_ be using it everywhere.

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,402 @@
+private refreshChild() {
+	if(childDevices){
+		def childDni = "${device.deviceNetworkId}:2"
+		def childDevice = childDevices.find { it.deviceNetworkId == childDni }
```
you could just do `childDevices?.find` instead of wrapping this in an if statement

or (probably better) pull out the finding of the child device into a function

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,402 @@
+					element: "paragraph",
+					description: "Option enabled: ${it.activeDescription}\n" + 
+						"Option disabled: ${it.inactiveDescription}" 
+				)
```
just put each of these input blocks on one line

I'd rather they be very long than clutter the DTH vertically

MGoralczykS committed:
Fixes

MGoralczykS commented:
@greens I fiexd those things that you have mentioned. Could you check it?

greens commented on devicetypes/smartthings/qubino-flush-1d-relay.src/qubino-flush-1d-relay.groovy:
```diff
@@ -0,0 +1,377 @@
+
+	if(!childDevice) {
+		log.debug "handleChildEvent / creating a child device"
+		childDevice = createChildDevice("qubino", "Child Temperature Sensor", childDni, "Qubino Temperature Sensor")
```
We had to rename this DTH to "Qubino Temperature Sensor"

greens replied:
I don't know how you're not seeing this comment.

MGoralczykS committed:
Fixes and add Qubino 1 Relay

MGoralczykS committed:
Fixes

MGoralczykS commented:
I created a new file Qubino Flush Relay to have there two Qubino Relays 1D and 1. I added 3 settings only for new Relay. Also changed those things that you wanted. Could you check it?

greens reviewed (changes requested):
You have to change the name of the DTH used to create the child device to "Qubino Temperature Sensor". "Child Temperature Sensor" does not exist in the "qubino" namespace.

MGoralczykS committed:
Change namespace name

MGoralczykS commented:
@greens Fixed

greens commented on devicetypes/smartthings/qubino-flush-relay.src/qubino-flush-relay.groovy:
```diff
@@ -0,0 +1,436 @@
+
+	if(!childDevice) {
+		log.debug "handleChildEvent / creating a child device"
+		childDevice = createChildDevice("smartthings", "Qubino Temperature Sensor", childDni, "Qubino Temperature Sensor")
```
This isn't the one I wanted you to change.

"Qubino Temperature Sensor" is in the "qubino" namespace.

"Child Temperature Sensor" is in the "smartthings" namespace.

You want to use "qubino" "Qubino Temperature Sensor"

MGoralczykS committed:
Change namespace

MGoralczykS commented:
@greens Fixed

greens commented:
No. It's not. You haven't addressed _any_ of my comments on `qubino-flush-1d-relay.groovy`

MGoralczykS commented:
@greens In my local repo I deleted qubino-flush-1d-relay because those Relays will be in one DTH.

greens commented:
@MGoralczykS you needed to `git rm` it. It still shows up in this PR.

MGoralczykS committed:
delete qubino-flush-1d-relay file

MGoralczykS commented:
@greens Done

greens commented on devicetypes/smartthings/qubino-flush-relay.src/qubino-flush-relay.groovy:
```diff
@@ -0,0 +1,436 @@
+		capability "Switch"
+		capability "Refresh"
+		capability "Sensor"
+		capability "Temperature Measurement"
```
this device shouldn't support temperature measurement since that's entirely on the child device.

greens commented on devicetypes/smartthings/qubino-flush-relay.src/qubino-flush-relay.groovy:
```diff
@@ -0,0 +1,436 @@
+	],
+	//Only for Qubino 1 Relay
+	[ 
+		name: "Input 3 contact type", key: "input3ContactType", type: "boolean",
```
So, if we're not surfacing the switches attached to input 1 and 3 at all anyway, why are these preferences here? 

I thought we were just going to add them as simple switch (or contact sensor) child devices?

greens commented:
Can you remind me why this is a separate DTH (as you've implemented it) and not part of `flush-2-relay`

MGoralczykS commented:
@greens When I started working on this device we didn't have `flush-2-relay`. I will add these devices to `flush-2-relay`.

MGoralczykS commented:
@greens Could you check this PR -> https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/37124 ?

greens commented:
Yeah. You can close this one.
